crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -1,40 +1,183 @@
+//! **Update:** Once an entity is created, you will need its [`Entity`] id to progress its life cycle.
+//! This can be done through [`World::entity_mut`](crate::world::World::entity_mut) and [`Commands::entity`](crate::system::Commands::entity).
+//! Even if you don't store the id, you can still find the entity you spawned by searching for it in a [`Query`].
+//! Queries are also the primary way of interacting with an entity's components.
+//! You can use [`EntityWorldMut::remove`](crate::world::EntityWorldMut::remove) and [`EntityCommands::remove`](crate::system::EntityCommands::remove) to remove components,
+//! and you can use [`EntityWorldMut::insert`](crate::world::EntityWorldMut::insert) and [`EntityCommands::insert`](crate::system::EntityCommands::insert) to insert more components.
+//! Be aware that each entity can only have 0 or 1 values for each kind of component, so inserting a bundle may overwrite existing component values.
+//! This can also be further configured based on the insert method.
+//!
+//! **Despawn:** Despawn an entity when it is no longer needed.
+//! This destroys it and all its components.
+//! The entity is no longer reachable through the [`World`], [`Commands`], or [`Query`]s.
+//! Note that this means an [`Entity`] id may refer to an entity that has since been despawned!
+//! Not all [`Entity`] ids refer to active entities.
+//! If an [`Entity`] id is used when its entity has been despawned, an [`EntityNotSpawnedError`] is emitted.
+//! Any [`System`](crate::system) could despawn any entity; even if you never share its id, it could still be despawned unexpectedly.
+//! Your code should do its best to handle these errors gracefully.
+//!
+//! In short:
+//!
+//! - Entities are spawned through methods like [`World::spawn`](crate::world::World::spawn), which return an [`Entity`] id for the new entity.
+//! - Once spawned, they can be accessed and modified through [`Query`]s and other apis.
+//! - You can get the [`Entity`] id of an entity through [`Query`]s, so losing an [`Entity`] id is not a problem.
+//! - Entities can have components inserted and removed via [`World::entity_mut`](crate::world::World::entity_mut) and [`Commands::entity`](crate::system::Commands::entity).
+//! - Entities are eventually despawned, destroying the entity and causing its [`Entity`] id to no longer refer to an entity.
+//! - Not all [`Entity`] ids point to actual entities, which makes many entity methods fallible.
+//!
+//! # Entity Ids
+//!
+//! As mentioned entities each have an [`Entity`] id, which is used to interact with that entity.
+//! But what actually is this id?
+//! This [`Entity`] id is the combination of two ideas: [`EntityRow`] and [`EntityGeneration`].
+//! You can think of the [`Entity`] type as a `struct Entity { row: u32, generation: u32 }`.
+//!
+//! To understand these ids, picture the ECS [`World`] as a spreadsheet.
+//! Each kind of component is represented by a column in the spreadsheet and each entity is a row.
```

Comment:
I don't like this phrasing because it makes it seem like the `Entity` is the "row index" for the table the components are stored in, when this is not the case. Even in a world where we stick with `Entity::row` terminology, I'd want to rephrase this. But as mentioned, I think we should embrace "entity index" terminology. And I think these docs should provide true functional descriptions of the system, rather than lossy abstractions.